dev/puppet-server.conf.sample

```diff
@@ -65,6 +65,18 @@ http-client: {
     #                TLS_RSA_WITH_AES_256_CBC_SHA,
     #                TLS_RSA_WITH_AES_128_CBC_SHA256,
     #                TLS_RSA_WITH_AES_128_CBC_SHA]
+
+    # The amount of time, in milliseconds, that an outbound HTTP connection
+    # will wait for no data to be available before closing the socket. If not
+    # defined, defaults to 20 minutes. If 0, the timeout is infinite and if
+    # negative, the value is undefined (system default).
```

Comment:
If you end up back in here, maybe an improvement might be, "is undefined in the application, the behavior is governed by the system default behavior"

It's a little confusing to me as it currently stands.

It's also worth asking...  Are we talking about a socket in the connected state, or are we talking about the timeout associated with establishing a TCP connection before it enters the connected state?

If we're talking about the former, a connected socket should ideally persist indefinitely even though it's idle with no data flowing.